### ISSUE TITLE
[Feature request] Show the type that violated the hint in error messages

### ISSUE THREAD

**Issue by justinchuby** [beartype/beartype#160]
I have seen values whose repr does not show the type of itself (bad) with which determining what caused the violation becomes trickier.

An example:

```
beartype.roar.BeartypeCallHintReturnViolation: @beartyped _run_symbolic_function() return [9 defined in (%9 : Float(*, *, *, strides=[12, 4, 1], device=cpu) = onnx::If(%15)
  block0()... violates type hint typing.Union[torch.Value, typing.Tuple[torch.Value, ...], NoneType], as [9 defined in (%9 : Float(*, *, *, strides=[12, 4, 1], device=cpu) = onnx::If(%15)
  block0()... not tuple, <protocol "builtins.NoneType">, or <protocol "torch.Value">.
```

It would be helpful for `beartype` to print out the offending type so that it is clear what is going on.

**Comment by leycec** [comment]
**Spot on.** Yet again, @justinchuby is absolutely right. This is all my fault, because [I actually resolved this a few months ago in 8cc89be for issue #124](https://github.com/beartype/beartype/issues/124#issuecomment-1100591599): e.g.,

```python
>>> from beartype import beartype
>>> @beartype
>>> def func(a: int) -> int:
...     return a
>>> func(5.)
beartype.roar.BeartypeCallHintParamViolation:
@beartyped func() parameter a=5.0 violates
type hint <class 'int'>, as float 5.0 not instance of int.
```

Is that the sort of type verbosity in violation messages that you and PyTorch alike are looking for? If so, this is your lucky Thursday! @leycec recommends buying a random number of lottery tickets.

In other words, I *desperately* need to push out our upcoming `beartype 0.11.0` release. Like, now. When critical issues begin recursively reappearing on the issue tracker, you know it's time to push. Thankfully, I'm nearly there. I'm currently finalizing support for class decoration at issue #152 – our last feature request this release cycle. :relieved: 

Let's see if I can excrete `beartype 0.11.0` tomorrow (Friday) evening. Don't fail me now, GitHub!

**Comment by justinchuby** [comment]
Yes! This is going to be very helpful. Something to potentially consider is also visual separation between beartype’s text and the user input, since for longer inputs it can be hard to identify when the user input starts. In my example above, the return value is a list. Since generic containers are also using `[]`, it took me a while to find out the list is what violates the hint. 

If I remember correctly, `None` was also quoted `”None”` when reported making me think a string was passed in, but this should be solved with the type message already.

**Comment by leycec** [comment]
> Something to potentially consider is also visual separation between beartype’s text and the user input...

**Definitely.** Human-readable violations is an ongoing Sisyphean struggle. Clearly delineating and demarcating parameter names from parameter types and values from return types and values from type hints is something we're still grappling with. If you think up any concrete ideas or examples of how @beartype might further improve that for you, would you mind submitting a new feature request?

And thanks so much. You've been a tremendous help with everything, Justin. Group hug! :hugs: :people_hugging: :hugs:
